### ISSUE TITLE
testOpenSSLBuffering fails intermittently on a dapper machine (running in vmware) because it uses reactor.iterate

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2374]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#2374|
|Type|defect|
|Created|2007-01-08 22:46:25Z|
|Branch|https://github.com/twisted/twisted/tree/test-ssl-iterate-2374|

```
Test Pass 128
Running 1 tests.
twisted.test.test_ssl
  BufferingTestCase
    testOpenSSLBuffering ...                                             [FAIL]

===============================================================================
[FAIL]: twisted.test.test_ssl.BufferingTestCase.testOpenSSLBuffering

Traceback (most recent call last):
  File "/home/radix/usr/lib/python2.4/site-packages/twisted/test/test_ssl.py", line 369, in testOpenSSLBuffering
    self.assertEquals(client.buffer, ["+OK <some crap>\r\n"])
twisted.trial.unittest.FailTest: [] != ['+OK <some crap>\r\n']
-------------------------------------------------------------------------------
```

The test uses reactor.iterate in a super lame way, and if I change the following line:

        while i < 5000 and not client.buffer:

To have a larger upper bound (say, 50,000) it very reliably passes. It shouldn't be doing iterate explicitly.

<details><summary>Searchable metadata</summary>

```
trac-id__2374 2374
type__defect defect
reporter__radix radix
priority__normal normal
milestone__ 
branch__branches_test_ssl_iterate_2374 branches/test-ssl-iterate-2374
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1168296385000000 1168296385000000
changetime__1211900230000000 1211900230000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [23461]) Branching to 'test-ssl-iterate-2374'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

This should close [#2918](https://github.com/twisted/twisted/issues/2918) too.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @therve|
|-|-|

`_runTest` could use a docstring.  Other than that, looks good.  Merge when ready.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [23717]) Merge test-ssl-iterate-2374

Author: therve
Reviewer: exarkun
Fixes [#2374](https://github.com/twisted/twisted/issues/2374)
Fixes [#2918](https://github.com/twisted/twisted/issues/2918)

Remove reactor.iterate calls from test_ssl.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
